### ISSUE TITLE
false positive `reportInvalidTypeVarUse` in return type position

### ISSUE THREAD

**Issue by KotlinIsland** [DetachHead/basedpyright#1489]
### Description

```py
from __future__ import annotations


class Box[T]:
    def __init__(self, t: T):
        self.t = t

    def copy[R](self) -> Box[R | T]:  # TypeVar "R" appears only once in generic function signature
        return Box(self.t)

class A: ...
class B(A): ...

b = Box(B())
b2: Box[A] = b.copy()
```

this rule has never made any sense to me, why on earth would using the return expectation ever be invalid?

**Comment by DetachHead** [comment]
simpler example:

Code sample in [basedpyright playground](https://basedpyright.com/?typeCheckingMode=all&code=CYUwZgBCC2AOAuBPA%2BgGwJYGd4G0AqAugBQCUEAtAHwQbb4EBcEAxFAE5sD2bTbIs3eAEkAdgDcAhhmB5EsEADUJbAKqYQAWABQEXRD7wArmxEQcBbdrCdOTWrnQj4BCAF4ocJGizxSLCJLSADQQAO6cIgDk8GHcANZh6PAAFpyGMSkgEADmICIgbOgAxkA)

```python
def empty_list[T]() -> list[T]: # error: reportInvalidTypeVarUse
    return []

foo: list[int] = empty_list() # valid, won't work without the generic
```
